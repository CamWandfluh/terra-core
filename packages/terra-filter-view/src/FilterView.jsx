import React from 'react';
import PropTypes from 'prop-types';
import 'terra-base/lib/baseStyles';
import classNames from 'classnames/bind';
import Select from './Select';
import { Variants } from './_constants';
import styles from './FilterView.scss';

const cx = classNames.bind(styles);


const propTypes = {
  /**
   * The dropdown/result menu options.
   */
  children: PropTypes.node,
  /**
   * The default selected value.
   */
  defaultValue: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.array]),
  /**
   * Whether the select is disabled.
   */
  disabled: PropTypes.bool,
  /**
   * When true, will disable the auto-search.
   */
  disableAutoSearch: PropTypes.bool,
  /**
   * Additional attributes to spread onto the dropdown. ( style, classnames, etc.. )
   */
  dropdownAttrs: PropTypes.object,
  /**
   * The minimum number of characters to perform a search.
   */
  minimumSearchTextLength: PropTypes.number,
  /**
   * Content to display when no results are found.
   */
  noResultContent: PropTypes.node,
  /**
   * Callback function triggered when the value changes. function(value)
   */
  onChange: PropTypes.func,
  /**
   * Callback function triggered when an option is selected.
   */
  onSelect: PropTypes.func,
  /**
   * A callback to indicate an invalid search.
   */
  onInvalidSearch: PropTypes.func,
  /**
   * Callback function triggered when the search criteria changes. function(searchvalue)
   */
  onSearch: PropTypes.func,
  /**
   * Callback function invoked for each option on search change. function(searchvalue, option)
   */
  optionFilter: PropTypes.func,
  /**
   * Placeholder text.
   */
  placeholder: PropTypes.string,
  /**
   * How long the component should wait (in milliseconds) after input before performing an automatic search.
   */
  searchDelay: PropTypes.number,
  /**
   * The search field variant. One of `dropdown` or `persistent`
   */
  variant: PropTypes.oneOf([
    Variants.DROPDOWN,
    Variants.PERSISTENT,
  ]),
};

const defaultProps = {
  children: undefined,
  defaultValue: undefined,
  disabled: false,
  disableAutoSearch: false,
  dropdownAttrs: undefined,
  minimumSearchTextLength: 2,
  noResultContent: undefined,
  onChange: undefined,
  onInvalidSearch: undefined,
  onSelect: undefined,
  optionFilter: undefined,
  placeholder: undefined,
  searchDelay: 250,
  variant: Variants.DROPDOWN,
};

const contextTypes = {
  /* eslint-disable consistent-return */
  intl: (context) => {
    if (context.intl === undefined) {
      return new Error('please add locale prop to base component to load translations');
    }
  },
};


function FilterView(props) {
  const FilterViewClassNames = cx([
    'filter-view',
  ]);

  const {
    children,
    defaultValue,
    disableAutoSearch,
    isBlock,
    onChange,
    onSelect,
    placeholder,
    variant,
    searchDelay,
    ...otherProps } = props;
  return (
    <Select
      className={FilterViewClassNames}
      id="search"
      defaultValue={defaultValue}
      onChange={onChange}
      onSelect={onSelect}
      placeholder={placeholder}
      searchDelay={searchDelay}
      variant={variant}
      {...otherProps}
    >
      {children}
    </Select>
  );
}


FilterView.propTypes = propTypes;
FilterView.defaultProps = defaultProps;
FilterView.contextTypes = contextTypes;
FilterView.Option = Select.Option;
FilterView.OptGroup = Select.OptGroup;

export default FilterView;
