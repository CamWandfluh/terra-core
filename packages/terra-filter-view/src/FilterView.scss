:local {
  .filter-view {
    /* TODO: Add styles */
  }
}
