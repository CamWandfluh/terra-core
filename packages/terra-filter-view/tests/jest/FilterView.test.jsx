import React from 'react';
import FilterView from '../../src/FilterView';
import intlContexts from './intl-context-setup';

describe('Snapshots', () => {
  it('renders a default FilterView', () => {
    const wrapper = shallow(<FilterView />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list FilterView', () => {
    const wrapper = shallow(<FilterView variant="list" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a default FilterView with a placeholder', () => {
    const wrapper = shallow(<FilterView placeholder="Test" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list FilterView with a placeholder', () => {
    const wrapper = shallow(<FilterView variant="list" placeholder="Test" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a default FilterView with a default value', () => {
    const wrapper = shallow(<FilterView defaultValue="Test" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a list FilterView with a default value', () => {
    const wrapper = shallow(<FilterView variant="list" defaultValue="Test" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a disabled default FilterView', () => {
    const wrapper = shallow(<FilterView disabled />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a disabled list FilterView', () => {
    const wrapper = shallow(<FilterView variant="list" disabled />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a disabled default FilterView with a default value', () => {
    const wrapper = shallow(<FilterView disabled defaultValue="default" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders a disabled list FilterView with a default value', () => {
    const wrapper = shallow(<FilterView variant="list" disabled defaultValue="default" />, intlContexts.shallowContext);
    expect(wrapper).toMatchSnapshot();
  });
  // Prop Tests
});
