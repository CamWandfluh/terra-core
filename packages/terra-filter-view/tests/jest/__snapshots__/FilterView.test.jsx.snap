// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots renders a default FilterView 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="dropdown"
/>
`;

exports[`Snapshots renders a default FilterView with a default value 1`] = `
<Select
  className="filter-view"
  defaultValue="Test"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="dropdown"
/>
`;

exports[`Snapshots renders a default FilterView with a placeholder 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  placeholder="Test"
  searchDelay={250}
  variant="dropdown"
/>
`;

exports[`Snapshots renders a disabled default FilterView 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={true}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="dropdown"
/>
`;

exports[`Snapshots renders a disabled default FilterView with a default value 1`] = `
<Select
  className="filter-view"
  defaultValue="default"
  disableAutoSearch={false}
  disabled={true}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="dropdown"
/>
`;

exports[`Snapshots renders a disabled list FilterView 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={true}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="list"
/>
`;

exports[`Snapshots renders a disabled list FilterView with a default value 1`] = `
<Select
  className="filter-view"
  defaultValue="default"
  disableAutoSearch={false}
  disabled={true}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="list"
/>
`;

exports[`Snapshots renders a list FilterView 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="list"
/>
`;

exports[`Snapshots renders a list FilterView with a default value 1`] = `
<Select
  className="filter-view"
  defaultValue="Test"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  searchDelay={250}
  variant="list"
/>
`;

exports[`Snapshots renders a list FilterView with a placeholder 1`] = `
<Select
  className="filter-view"
  disableAutoSearch={false}
  disabled={false}
  id="search"
  isInvalid={false}
  minimumSearchTextLength={2}
  placeholder="Test"
  searchDelay={250}
  variant="list"
/>
`;
